Providers/Profile.swift

```diff
@@ -452,8 +462,8 @@ open class BrowserProfile: Profile {
         return SQLiteMetadata(db: self.database)
     }()
 
-    lazy var placesDbPath = URL(fileURLWithPath: (try! files.getAndEnsureDirectory()), isDirectory: true).appendingPathComponent("places.db").path
-    lazy var browserDbPath =  URL(fileURLWithPath: (try! self.files.getAndEnsureDirectory())).appendingPathComponent("browser.db").path
+    lazy var browserDbPath = URL(fileURLWithPath: directory).appendingPathComponent("browser.db").path
```

Comment:
@yoanarios @atfelix after some investigation, the issue with the tests comes from the changes in this file. For some reason this is breaking the test in the Onboarding test suite and the Search tests that you see failing in this PR.

I undid these changes and all the tests pass.

Reply:
Thank you for finding the issue.

Are the test failures occurring locally or only on CI?  If they're failing locally, I can investigate after work (5 PM Toronto time).

Reply:
They happen locally too.
If you select SmokeXCUITests Test Plan, the OnboardingTests suite will be enable, run it and you will see the failures.
Same for the SearchTests suite, but using the Smoketest4 as the TestPlan

Hope that helps. Thanks for taking the time to look at this.

Reply:
I think I have a fix for this issue.  I believe (but it's hard to know for sure) that the `clear` in `BrowserProfile.init` and where I put the `directory` do block is causing the issue.  I have moved the `directory` to a lazy variable that crashes via a `fatalError` in the `catch` block (same behaviour as current production).  This move seems to fix the issue but it's difficult to say as running the tests and getting consistent results seems difficult (at least locally).

Please let me know if there are any other CI issues.